lib/SIL/Utils/OptimizationRemark.cpp

```diff
@@ -100,6 +103,93 @@ Emitter::Emitter(StringRef passName, SILModule &m)
           m.getASTContext().LangOpts.OptimizationRemarkMissedPattern->match(
               passName)) {}
 
+/// The user has passed us an instruction that for some reason has a source loc
+/// that can not be used. Search down the current block for an instruction with
+/// a valid source loc and use that instead.
+static SourceLoc inferOptRemarkSearchForwards(SILInstruction &i) {
+  for (auto &inst :
+       llvm::make_range(std::next(i.getIterator()), i.getParent()->end())) {
+    auto newLoc = inst.getLoc().getSourceLoc();
+    if (newLoc.isValid())
+      return newLoc;
+  }
+
+  return SourceLoc();
+}
+
+/// The user has passed us an instruction that for some reason has a source loc
+/// that can not be used. Search up the current block for an instruction with
+/// a valid SILLocation and use the end SourceLoc of the SourceRange for the
+/// instruction.
+static SourceLoc inferOptRemarkSearchBackwards(SILInstruction &i) {
+  for (auto &inst : llvm::make_range(std::next(i.getReverseIterator()),
+                                     i.getParent()->rend())) {
+    auto loc = inst.getLoc();
+    if (!bool(loc))
+      continue;
+
+    auto range = loc.getSourceRange();
+    if (range.isValid())
+      return range.End;
+  }
+
+  return SourceLoc();
+}
+
+SourceLoc swift::OptRemark::inferOptRemarkSourceLoc(
+    SILInstruction &i, SourceLocInferenceBehavior inferBehavior)
+    __attribute__((optnone)) {
```

Comment:
Oops. I need to remove this. I'm going to remove it after this finishes testing.